core/src/main/scala/com/softwaremill/react/kafka2/Consumer.scala

```diff
@@ -75,7 +89,7 @@ object ManualCommitConsumer {
   }
 }
 
-class ManualCommitConsumer[K, V](consumerProvider: ConsumerProvider[K, V])
+class ManualCommitConsumer[K, V](consumerProvider: () => KafkaConsumer[K, V])
```

Comment:
I dont know way to manually assign partition through the properties. Same for offsets for topics/partitions (in case of manual external offset tracking). Could you point me to way to do this?

Mocking `KafkaConsumer` is useful for testing reactive-kafka itself. Like all tests in this commit :) In case of explicit creation of `KafkaConsumer` I dont know how to test all these cases of commit fails, empty poll responses and so.